packages/runtime-dom/src/modules/events.ts

```diff
@@ -43,19 +49,39 @@ const getNow = () => cachedNow || (p.then(reset), (cachedNow = _getNow()))
 export function patchEvent(
   el: Element,
   name: string,
-  prevValue: EventValue | null,
-  nextValue: EventValue | null,
+  prevValue: EventValueWithOptions | EventValue | null,
+  nextValue: EventValueWithOptions | EventValue | null,
   instance: ComponentInternalInstance | null = null
 ) {
+  const prevOptions = prevValue && 'options' in prevValue && prevValue.options
+  const nextOptions = nextValue && 'options' in nextValue && nextValue.options
   const invoker = prevValue && prevValue.invoker
-  if (nextValue) {
+  const value =
+    nextValue && 'handler' in nextValue ? nextValue.handler : nextValue
+
+  // go unoptimized route if it has to deal with event options
```

Comment:
I don't have a clear idea on how to optimize this yet, but this is definitely too unoptimized - it will cause the event listener to be detached and re-attached on every single component update.

Also need to consider compiled case - for example with `@click.cpature` we know the options will never change so it should not need to go this route.

Reply:
> I don't have a clear idea on how to optimize this yet, but this is definitely too unoptimized - it will cause the event listener to be detached and re-attached on every single component update.

I thought about `looseEqual` check and thought it would add more overhead than event re-assignment.

> Also need to consider compiled case - for example with @click.cpature we know the options will never change so it should not need to go this route.

Could this be a compiler flag?
```
nextValue = {
  handler: () => {},
  options: {},
  persistent: true
}
```

Reply:
To my surprise `looseEqual` check is much faster than event re-assignment. So probably not that bad of a tradeoff.

https://jsperf.com/looseequal-or-event-re-assign/1